src/lms-ds-loader/edfi_lms_ds_loader/scripts/mssql/add_mapping_columns_for_edfi_student_and_section.sql

```diff
@@ -0,0 +1,20 @@
+-- SPDX-License-Identifier: Apache-2.0
+-- Licensed to the Ed-Fi Alliance under one or more agreements.
+-- The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+-- See the LICENSE and NOTICES files in the project root for more information.
+
+ALTER TABLE lms.LMSSection ADD EdFiSectionId UNIQUEIDENTIFIER NULL;
+
+ALTER TABLE lms.LMSUser ADD EdFiStudentId UNIQUEIDENTIFIER NULL;
+
+CREATE UNIQUE NONCLUSTERED INDEX UX_LMSSection_EdfiSectionId
+    ON lms.LMSSection (EdFiSectionId)
+    WHERE EdFiSectionId is NOT NULL;
```

Comment:
Still researching this, as it might be used in an update statement. And we might want the deletedat column too.

Reply:
I created four different indexes, none of which were used the by query above (and *none of which are unique* because I fear that one SIS section might somehow end up with multiple LMS sections):

```sql
CREATE NONCLUSTERED INDEX UX_LMSSection_EdfiSectionId
    ON lms.LMSSection (EdFiSectionId)
    WHERE EdFiSectionId is NOT NULL;

CREATE NONCLUSTERED INDEX UX_LMSSection_EdfiSectionId_2
    ON lms.LMSSection (EdFiSectionId);
        
CREATE NONCLUSTERED INDEX UX_LMSSection_EdfiSectionId_3
    ON lms.LMSSection (EdFiSectionId)
    WHERE EdFiSectionId is NOT NULL AND DeletedAt IS NOT NULL;

CREATE NONCLUSTERED INDEX UX_LMSSection_EdfiSectionId_4
    ON lms.LMSSection (EdFiSectionId)
    WHERE DeletedAt IS NOT NULL;
```

Then I looked at the query plan for the following update statement:

```sql
update lms set
    EdFiSectionId = s.Id
from lms.LMSSection lms
inner join edfi_ods_glendale_v50.edfi.Section s
    on lms.SisSectionIdentifier = s.SectionIdentifier
where lms.EdFiSectionId  is null
    and lms.DeletedAt is not null
```

The fourth index was used. Please try it out on your system too to confirm.